### PR TITLE
tic80: init at 1.0.2164

### PR DESCRIPTION
**Pull request by CyborgPotato**
###### Description of changes

Created TIC-80 package, as requested by #200260.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/tic80/default.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+, Cocoa
+, CoreAudio
+, CoreServices
+, ForceFeedback
+, OpenGL
+}:
+
+stdenv.mkDerivation rec {
+  pname = "tic-80";
+
```
Comment (azahi) [inline pkgs/development/tools/tic80/default.nix:18]:
```suggestion
```

File: pkgs/development/tools/tic80/default.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    fetchSubmodules = true;
+    sha256 = "SnaAhdYoblxKlzTSyfcnvY6u7X6aIJIYWZAXtL2IIXc=";
+  };
+
+  meta = with lib; {
+    description = "TIC-80 is a free and open source fantasy computer for making, playing and sharing tiny games.";
+    homepage = "https://tic80.com";
+    license = licenses.mit;
+    platforms = platforms.all;
+    maintainers = with maintainers; [ ];
```
Comment (azahi) [inline pkgs/development/tools/tic80/default.nix:58]: You should include yourself as a maintainer.

File: pkgs/development/tools/tic80/default.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    xcbuild
+  ] ++ [
+    AVFoundation
+    AudioUnit
+    Cocoa
+    CoreAudio
+    CoreServices
+    ForceFeedback
+    OpenGL
+  ]);
```
Comment (azahi) [inline pkgs/development/tools/tic80/default.nix:38]: It's probably should look more like this:
```suggestion
  nativeBuildInputs = [
    cmake
    ruby
  ] ++ lib.optionals stdenv.isDarwin [
    xcbuild
  ];

  buildInputs = lib.optionals stdenv.isDarwin [
    AVFoundation
    AudioUnit
    Cocoa
    CoreAudio
    CoreServices
    ForceFeedback
    OpenGL
  ];
```

File: pkgs/development/tools/tic80/default.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  postInstall =
+    if stdenv.isDarwin then ''
+      cp -r bin $out/
+    '' else "";
```
Comment (azahi) [inline pkgs/development/tools/tic80/default.nix:43]:
```suggestion
  postInstall = lib.optionalString stdenv.isDarwin ''
    cp -r bin $out
  '';
```

File: pkgs/development/tools/tic80/default.nix [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+  src = fetchFromGitHub {
+    owner = "nesbox";
+    repo = "TIC-80";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "SnaAhdYoblxKlzTSyfcnvY6u7X6aIJIYWZAXtL2IIXc=";
+  };
+
+  meta = with lib; {
+    description = "TIC-80 is a free and open source fantasy computer for making, playing and sharing tiny games.";
```
Comment (azahi) [inline pkgs/development/tools/tic80/default.nix:54]:
```suggestion
    description = "TIC-80 is a free and open source fantasy computer for making, playing and sharing tiny games";
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/tic80/default.nix (hunk 2 above)
Comment (DavidCromp) [reply pkgs/development/tools/tic80/default.nix:57]: Good point, I was waiting for another PR where I added myself to the maintainers list, but good point. Will do

**Comment by DavidCromp** [comment]
Found issues that need to be resolved, as well as further SDL build options etc. for static builds

**Pull request closed without merging**